content-watcher.js

```diff
@@ -34,31 +36,52 @@ function sendPing() {
 }
 
 function putUpdate(path) {
-  const data = JSON.stringify({
-    relative_path: path,
-    contents: fs.readFileSync(path).toString()
-  })
-  const options = postRequestOptions(
-    `/attendances/${process.env.ATTENDANCE_ID}/file_snapshots`,
-    {
-      "Content-Type": "application/json",
-      "Content-Length": data.length
-    }
-  )
+  if(!changed_files.includes(path)) {
+    changed_files.push(path)
+  }
+}
 
-  const req = https.request(options, res => {
-    console.log(`status: ${res.statusCode}`)
+function uploadChangedFiles() {
+  // Iterate over a copy so the indexes don't change as we modify the array
+  changed_files_copy = [...changed_files]
 
-    res.on("data", d => {
-      process.stdout.write(d)
+  changed_files_copy.forEach(function(path) {
+    process.stdout.write(`uploading: ${path}\n`)
```

Comment:
This `forEach` block ends up being quite long - just for readability, I'd probably want to pull it out into a separate function. Up to you though.